### PR TITLE
[RDY] Bug fix: check if evaluation method actually exists

### PR DESCRIPTION
**Pull request by Balletie**

### CODE REVIEW

File: org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/functions/EvaluateFunction.java [hunk 1]
```diff
@@ -55,14 +56,33 @@ public EvaluateFunction(Map<String, IEvaluationStrategy> evaluationStrategies,
...
+        if (facet == null) {
+            throw new MetaborgException("No ESV configuration found for the REPL.");
+        }
+        return facet;
+    }
+
+    private IEvaluationStrategy evaluationStrategy(String evaluationMethod)
+        throws MetaborgException {
+        IEvaluationStrategy evalStrategy = evaluationStrategies.get(evaluationMethod);
+        Iterable<String> quoted = evaluationStrategies.keySet().stream().map(e -> '\"' + e + '\"')
```
Comment (Hjdskes) [inline org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/functions/EvaluateFunction.java]: Why don't you construct this only when required, e.g. inside the `if`?

**Comment by Hjdskes** [comment]
```
[ERROR] /home/travis/build/spoofax-shell/spoofax-shell/org/metaborg/spoofax/shell/core/src/main/java/org/metaborg/spoofax/shell/core/DynSemEvaluationStrategy.java:[136] The method makeKey(String, String, int) is undefined for the type RuleRegistry
```

Is that the compilation errors you were talking about yesterday?

**Comment by Balletie** [comment]
Yes that's the error. We will merge this PR after updating dynsem, so that we can get green _travis_ lights again ;).

File: org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/functions/EvaluateFunction.java (hunk 1 above)
Comment (Balletie) [inline org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/functions/EvaluateFunction.java]: good point, I will update this

**Pull request merged**